File: src/engine/component/transform.cpp
//
// Created by Nemo li on 2023/12/6.
//

#include "transform.h"
#include <rttr/registration>

//注册反射
RTTR_REGISTRATION {
    rttr::registration::class_<Transform>("Transform")
            .constructor<>()(rttr::policy::ctor::as_raw_ptr)
            .property("position", &Transform::position, &Transform::set_position)
            .property("rotation", &Transform::rotation, &Transform::set_rotation)
            .property("scale", &Transform::scale, &Transform::set_scale);
}

Transform::Transform() : position_(0.f), rotation_(0.f), scale_(1.f) {
}

Transform::~Transform() {
}

File: engine_lua/component/transform.lua
require("lua_extension")

Transform=class("Transform",Component)

function Transform:ctor()
    Transform.super.ctor(self)
end

function Transform:InitCppComponent()
    ---@type Cpp.Transform
    self.cpp_component_instance_=Cpp.Transform()
end


function Transform:position()
    return self.cpp_component_instance_:position()
end

function Transform:rotation()
    return self.cpp_component_instance_:rotation()
end

function Transform:scale()
    return self.cpp_component_instance_:scale()
end

function Transform:set_position(position)
    return self.cpp_component_instance_:set_position(position)
end

function Transform:set_rotation(position)
    return self.cpp_component_instance_:set_rotation(position)
end

function Transform:set_scale(position)
    return self.cpp_component_instance_:set_scale(position)
end